magento/pwa-studio
# PWA-2308: Update ConfigurableProductOptions GraphQL field from id to uid

- Updated peregrine/lib/talons/AddToCartDialog/addToCartDialog.gql.js



## Description
### Overview
The field ConfigurableProductOptions.id is deprecated. We should use the `uid` field instead.

By default, Apollo client uses the `__typename` and `id` field for caching so we will also need to update the fields that Apollo client uses to cache ConfigurableProductOptions objects from `id` to `uid`.

Found in:

peregrine/lib/talons/AddToCartDialog/addToCartDialog.gql.js
### Acceptance Criteria
PWA Studio no longer requests the deprecated `id` field on `ConfigurableProductOptions` objects
ConfigurableProductOptions objects are still cached properly

## Related Issue







[PWA-2308](https://jira.corp.magento.com/browse/PWA-2308)

## Acceptance





### Verification Stakeholders



### Specification




## Verification Steps
### Test Plan

1. Sign In
2. Add a configurable product to your wishlist without selecting any options on the product
3. Navigate to your wishlist
4. Attempt to add the product to your cart
5. Verify the Add to Cart dialog appears as expected
6. Click "Cancel"
7. Go offline
8. Attempt to add the product to your cart again
9. Verify that the Add to Cart dialog and all configurable options appear as expected

#### Test scenario(s) for direct fix/feature





#### Test scenario(s) for any existing impacted features/areas





#### Test scenario(s) for any Magento Backend Supported Configurations





#### Is Browser/Device testing needed?





#### Any ad-hoc/edge case scenarios that need to be considered?




## Screenshots / Screen Captures (if appropriate)

## Breaking Changes (if any)





## Checklist



-   I have added tests to cover my changes, if necessary.
-   I have added translations for new strings, if necessary.
-   I have updated the documentation accordingly, if necessary.
